experimental/benchmark/phoronix/images/ansible/roles/win-phoronix/README.md

```diff
@@ -0,0 +1,76 @@
+# ansible-win-phoronix
+
+This is a role for installing phoronix under windows.
+
+## Requirements
+
+
+  * KVM or similar
+  * Ansible (2.8+?)
+  * Compatible version of winrm for ansible
+
+## User Configurable variables
+
+  * winphoronix_shutdown: set to true to shut down at the end
+  * winphoronix_sysprep: set to true to run sysprep /generalize
+
+References:
+
+  * <https://docs.microsoft.com/en-us/windows-hardware/manufacture/desktop/sysprep--generalize--a-windows-installation>
+
+## Tags
+
+  * virtio
+  * xen
+
+## End to End setup
+
+### Download/run initial image
+
+```
+# Create and go into directory
+mkdir MSEdge.Win10
+cd MSEdge.Win10
+
+# Download from https://developer.microsoft.com/en-us/microsoft-edge/tools/vms/
+wget https://az792536.vo.msecnd.net/vms/VMBuild_20190311/VirtualBox/MSEdge/MSEdge.Win10.VirtualBox.zip
+
+# Unzip
+unzip ../MSEdge.Win10.VirtualBox.zip
+
+# Unpack
+mv "MSEdge - Win10.ova" MSEdge-Win10.tar
+tar xaf MSEdge-Win10.tar
+
+# Convert to QCOW2
+qemu-img convert "MSEdge - Win10-disk001.vmdk" -O qcow2 -S 4k -c MSEdge-Win10-disk001.qcow2
+
+# Create dummy virtio disk
+truncate -s 32M dummy.raw
+
+# Start under KVM with port redirection
+qemu-system-x86_64 -enable-kvm -smp cpus=2 -m 2048M -net nic \
+  -net user,hostfwd=tcp::5985-:5985,hostfwd=tcp::8022-:22 \
+  -drive file=MSEdge-Win10-disk001.qcow2 \
+  -drive file=dummy.raw,format=raw,if=virtio \
+  -monitor stdio
+```
+
+### Ubuntu Focal Packages
+
+  * ansible
+  * python3-winrm
+
+### Manual setup in Windows VM
+
+  * Log in using default credential IEUser / Passw0rd!
+  * Click on network connection
+  * Click on "network and internet settings"
+  * Click on "change connection properties"
+  * Select "private" not "public"
```

Comment:
We need to do it manually? Can we do it via ssh or something else?
Now, I get:
```
pts/fio-1.13.2:
        Test Installation 1 of 1
        1 File Needed [3.55 MB]
        Downloading: fio-3.18-x64.zip                                                                       [3.55MB]
            Download Failed: http://bsdio.com/fio/releases/fio-3.18-x64.zip
            Downloading of needed test files failed.

```

Reply:
> We need to do it manually? Can we do it via ssh or something else?

Yes and no. The reason to perform this manual step is to get WinRM https://docs.microsoft.com/en-us/windows/win32/winrm/portal to listen on the network interface, which is the service you need to automate everything else, including installing SSH. SSH is not installed at this step, that happens in the ansible playbook.

You might be able to get around this requirement by booting the VM in VirtualBox or a different image where the virtualization technology matches what MS had, if that means that the image comes up with an existing interface already on a private network.

> Now, I get:
> 
> ```
> pts/fio-1.13.2:
>         Test Installation 1 of 1
>         1 File Needed [3.55 MB]
>         Downloading: fio-3.18-x64.zip                                                                       [3.55MB]
>             Download Failed: http://bsdio.com/fio/releases/fio-3.18-x64.zip
>             Downloading of needed test files failed.
> ```

I don't know, it used to work. I might try starting by seeing if there's a newer phoronix version.